### ISSUE TITLE
cannot reproduce the results in the paper

### ISSUE THREAD

**Issue by smallcute1** [SongW-SW/X-FNC#2]
Hello: 
     After reading your paper, I have been greatly inspired and would like to thank you for your contributions to the paper. However, when I attempted to replicate the results, I was unable to achieve the accuracy levels reported within the paper. Therefore, I carefully examined the code and found a part that I could not understand: "attention=torch.exp((cos_similarity(pos_graph_feat, pos_graph_eat)*2-2)*100)". This might return a tensor representing the similarity between two vectors (which may be the same vector here), and it seems that all the values are 1. Below is the result I printed out:attention tensor([1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1.,
        1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1.,
        1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1.,
        1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1.,
        1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1.,
        1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1.,
        1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1.,
        1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1.,
        1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1.,
        1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 
        1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1.,
        1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1.]Regarding the attention mechanism, the paper describes it as follows: "we propose to compute another edge weight matrix based on Euclidean distances between node features as follows," which seems to have some discrepancies. It's possible that I may not have a clear understanding of it. Therefore, I modified that line of code to store the Euclidean distance between node features. After the change, on the cora-full dataset, with a setting of 5-way 3 shot and 10 labels available per class, the accuracy did indeed increase: "Final Test Acc:0.5788 Best Acc:0.5788 Pse_test acc: 0.4136", which is one of the test sets. Thus, I suspect that this line of code might be the reason for the lower accuracy, although there is still a gap compared to the accuracy in the paper.
    And,During my replication process, I noticed that as the number of training epochs increased, the accuracy continued to decline. Here are some examples:0.2840  Best Acc: 0.6320 Pse_test acc: 0.5183 
Epoch: 2001 loss_train: 0.1744 acc_train: 1.0000
Epoch: 2021 loss_train: 0.7870 acc_train: 0.6000
Epoch: 2041 loss_train: 0.2540 acc_train: 0.8000
Epoch: 2061 loss_train: 0.7750 acc_train: 0.6000
Epoch: 2081 loss_train: 2.5778 acc_train: 0.6000
Final Test Acc: 0.2400  Best Acc: 0.6320 Pse_test acc: 0.5250 
Epoch: 2101 loss_train: 1.1557 acc_train: 0.4000
Epoch: 2121 loss_train: 1.5912 acc_train: 0.2000
Epoch: 2141 loss_train: 2.1310 acc_train: 0.2000
Epoch: 2161 loss_train: 26.8664 acc_train: 0.2000
Epoch: 2181 loss_train: 4.3613 acc_train: 0.0000
Final Test Acc: 0.2020  Best Acc: 0.6320 Pse_test acc: 0.5483 
Epoch: 2201 loss_train: 14.6306 acc_train: 0.2000
Epoch: 2221 loss_train: 4.8300 acc_train: 0.2000
Epoch: 2241 loss_train: 2.0180 acc_train: 0.2000
Epoch: 2261 loss_train: 4.2938 acc_train: 0.2000
Epoch: 2281 loss_train: 25.6772 acc_train: 0.2000
Final Test Acc: 0.1980  Best Acc: 0.6320 Pse_test acc: 0.4883 
Epoch: 2301 loss_train: 5.6008 acc_train: 0.2000
Epoch: 2321 loss_train: 1.9435 acc_train: 0.2000
Epoch: 2341 loss_train: 2.4558 acc_train: 0.2000
Epoch: 2361 loss_train: 2.2980 acc_train: 0.2000
Epoch: 2381 loss_train: 9.3132 acc_train: 0.2000
Final Test Acc: 0.1960  Best Acc: 0.6320 Pse_test acc: 0.5000 
Epoch: 2401 loss_train: 1.6362 acc_train: 0.2000
Epoch: 2421 loss_train: 2.2899 acc_train: 0.2000
Epoch: 2441 loss_train: 1.8634 acc_train: 0.2000
Epoch: 2461 loss_train: 1.7126 acc_train: 0.2000
Epoch: 2481 loss_train: 1.7811 acc_train: 0.2000
Final Test Acc: 0.2060  Best Acc: 0.6320 Pse_test acc: 0.5317 
Epoch: 2501 loss_train: 2.3992 acc_train: 0.2000
Epoch: 2521 loss_train: 1.6605 acc_train: 0.2000
Epoch: 2541 loss_train: 1.6882 acc_train: 0.2000
Epoch: 2561 loss_train: 1.7004 acc_train: 0.2000
Epoch: 2581 loss_train: 2.5342 acc_train: 0.2000
Final Test Acc: 0.2000  Best Acc: 0.6320 Pse_test acc: 0.5183 
Epoch: 2601 loss_train: 1.6838 acc_train: 0.2000
Epoch: 2621 loss_train: 1.8885 acc_train: 0.2000
Epoch: 2641 loss_train: 1.6924 acc_train: 0.2000
Epoch: 2661 loss_train: 1.6401 acc_train: 0.2000
Epoch: 2681 loss_train: 1.8592 acc_train: 0.2000
Final Test Acc: 0.2020  Best Acc: 0.6320 Pse_test acc: 0.4983 
Epoch: 2701 loss_train: 1.6175 acc_train: 0.2000
Epoch: 2721 loss_train: 1.6913 acc_train: 0.2000
Epoch: 2741 loss_train: 5.6927 acc_train: 0.2000
Epoch: 2761 loss_train: 3.1431 acc_train: 0.2000
Epoch: 2781 loss_train: 1.6743 acc_train: 0.2000
Final Test Acc: 0.2000  Best Acc: 0.6320 Pse_test acc: 0.4517 
Epoch: 2801 loss_train: 1.7136 acc_train: 0.2000
Epoch: 2821 loss_train: 9.3383 acc_train: 0.2000
Epoch: 2841 loss_train: 1.6743 acc_train: 0.2000
Epoch: 2861 loss_train: 1.7262 acc_train: 0.2000
Epoch: 2881 loss_train: 2.0138 acc_train: 0.2000
Final Test Acc: 0.2000  Best Acc: 0.6320 Pse_test acc: 0.5650 
Epoch: 2901 loss_train: 1.6768 acc_train: 0.2000
Epoch: 2921 loss_train: 1.6965 acc_train: 0.2000
Epoch: 2941 loss_train: 1.6691 acc_train: 0.2000
Epoch: 2961 loss_train: 2.2302 acc_train: 0.2000
Epoch: 2981 loss_train: 1.6614 acc_train: 0.2000
Final Test Acc: 0.2000  Best Acc: 0.6320 Pse_test acc: 0.5667 
Epoch: 3001 loss_train: 1.7262 acc_train: 0.2000
Epoch: 3021 loss_train: 1.6533 acc_train: 0.2000
Epoch: 3041 loss_train: 2.5890 acc_train: 0.2000
Epoch: 3061 loss_train: 1.7264 acc_train: 0.2000
Epoch: 3081 loss_train: 1.6476 acc_train: 0.2000
Final Test Acc: 0.2000  Best Acc: 0.6320 Pse_test acc: 0.4750 
Epoch: 3101 loss_train: 1.6687 acc_train: 0.2000
Epoch: 3121 loss_train: 1.6922 acc_train: 0.2000
Epoch: 3141 loss_train: 1.6419 acc_train: 0.2000
Epoch: 3161 loss_train: 5.0638 acc_train: 0.2000
Epoch: 3181 loss_train: 1.6477 acc_train: 0.2000
Final Test Acc: 0.2000  Best Acc: 0.6320 Pse_test acc: 0.4317 
Epoch: 3201 loss_train: 1.6923 acc_train: 0.2000
Epoch: 3221 loss_train: 1.7262 acc_train: 0.2000
Epoch: 3241 loss_train: 1.6929 acc_train: 0.2000
Epoch: 3261 loss_train: 1.6723 acc_train: 0.2000
Epoch: 3281 loss_train: 1.6687 acc_train: 0.2000
Final Test Acc: 0.2000  Best Acc: 0.6320 Pse_test acc: 0.4550 
Epoch: 3301 loss_train: 1.7046 acc_train: 0.2000
Epoch: 3321 loss_train: 1.6687 acc_train: 0.2000
Epoch: 3341 loss_train: 1.6476 acc_train: 0.2000
Epoch: 3361 loss_train: 1.6921 acc_train: 0.2000
Epoch: 3381 loss_train: 1.6923 acc_train: 0.2000
Final Test Acc: 0.2000  Best Acc: 0.6320 Pse_test acc: 0.5067 
Epoch: 3401 loss_train: 1.6475 acc_train: 0.2000
Epoch: 3421 loss_train: 1.6576 acc_train: 0.2000
Epoch: 3441 loss_train: 1.6922 acc_train: 0.2000
Epoch: 3461 loss_train: 1.6922 acc_train: 0.2000
Epoch: 3481 loss_train: 1.6922 acc_train: 0.2000
Final Test Acc: 0.2000  Best Acc: 0.6320 Pse_test acc: 0.5067 
Epoch: 3501 loss_train: 1.6477 acc_train: 0.2000
Epoch: 3521 loss_train: 1.6922 acc_train: 0.2000
Epoch: 3541 loss_train: 1.6922 acc_train: 0.2000
Epoch: 3561 loss_train: 1.6752 acc_train: 0.2000
Epoch: 3581 loss_train: 1.6476 acc_train: 0.2000
Final Test Acc: 0.1960  Best Acc: 0.6320 Pse_test acc: 0.5083 
Epoch: 3601 loss_train: 1.6687 acc_train: 0.2000
Epoch: 3621 loss_train: 1.6687 acc_train: 0.2000
Epoch: 3641 loss_train: 1.6745 acc_train: 0.2000
Epoch: 3661 loss_train: 1.6476 acc_train: 0.2000
Epoch: 3681 loss_train: 1.6922 acc_train: 0.2000
Final Test Acc: 0.2000  Best Acc: 0.6320 Pse_test acc: 0.4800 
Epoch: 3701 loss_train: 1.6476 acc_train: 0.2000
Epoch: 3721 loss_train: 1.6628 acc_train: 0.2000
Epoch: 3741 loss_train: 1.6949 acc_train: 0.2000
Epoch: 3761 loss_train: 1.6477 acc_train: 0.2000
Epoch: 3781 loss_train: 1.6625 acc_train: 0.2000
Final Test Acc: 0.2000  Best Acc: 0.6320 Pse_test acc: 0.5083 
Epoch: 3801 loss_train: 1.6476 acc_train: 0.2000
Epoch: 3821 loss_train: 1.6918 acc_train: 0.2000
Epoch: 3841 loss_train: 1.6752 acc_train: 0.2000
Epoch: 3861 loss_train: 1.6687 acc_train: 0.2000
Epoch: 3881 loss_train: 1.8136 acc_train: 0.2000
Final Test Acc: 0.2000  Best Acc: 0.6320 Pse_test acc: 0.5017 
Epoch: 3901 loss_train: 1.6760 acc_train: 0.2000
Epoch: 3921 loss_train: 1.6743 acc_train: 0.2000
Epoch: 3941 loss_train: 1.6924 acc_train: 0.2000
Epoch: 3961 loss_train: 1.6476 acc_train: 0.2000
Epoch: 3981 loss_train: 1.7261 acc_train: 0.2000
Final Test Acc: 0.2000  Best Acc: 0.6320 Pse_test acc: 0.5117 
Epoch: 4001 loss_train: 1.6744 acc_train: 0.2000
Epoch: 4021 loss_train: 1.6637 acc_train: 0.2000
Epoch: 4041 loss_train: 1.6922 acc_train: 0.2000
Epoch: 4061 loss_train: 1.6265 acc_train: 0.2000
Epoch: 4081 loss_train: 1.6743 acc_train: 0.2000
Final Test Acc: 0.2000  Best Acc: 0.6320 Pse_test acc: 0.4933 
Epoch: 4101 loss_train: 1.7262 acc_train: 0.2000
Epoch: 4121 loss_train: 1.7262 acc_train: 0.2000
Epoch: 4141 loss_train: 1.6924 acc_train: 0.2000
Epoch: 4161 loss_train: 1.6922 acc_train: 0.2000
Epoch: 4181 loss_train: 1.6922 acc_train: 0.2000
Final Test Acc: 0.2000  Best Acc: 0.6320 Pse_test acc: 0.4567 
Epoch: 4201 loss_train: 1.6743 acc_train: 0.2000
Epoch: 4221 loss_train: 1.7262 acc_train: 0.2000
Epoch: 4241 loss_train: 1.7262 acc_train: 0.2000
Epoch: 4261 loss_train: 1.6687 acc_train: 0.2000
Epoch: 4281 loss_train: 1.6923 acc_train: 0.2000
Final Test Acc: 0.2000  Best Acc: 0.6320 Pse_test acc: 0.4350 
Epoch: 4301 loss_train: 1.6476 acc_train: 0.2000
Epoch: 4321 loss_train: 1.6475 acc_train: 0.2000
Epoch: 4341 loss_train: 1.6744 acc_train: 0.2000
Epoch: 4361 loss_train: 1.7264 acc_train: 0.2000
Epoch: 4381 loss_train: 1.6923 acc_train: 0.2000
Final Test Acc: 0.2000  Best Acc: 0.6320 Pse_test acc: 0.4517 
Epoch: 4401 loss_train: 1.6476 acc_train: 0.2000
Epoch: 4421 loss_train: 1.6348 acc_train: 0.2000
Epoch: 4441 loss_train: 1.6687 acc_train: 0.2000
Epoch: 4461 loss_train: 1.6922 acc_train: 0.2000
Epoch: 4481 loss_train: 1.6750 acc_train: 0.2000
Final Test Acc: 0.2000  Best Acc: 0.6320 Pse_test acc: 0.5333 
Epoch: 4501 loss_train: 1.6743 acc_train: 0.2000
Epoch: 4521 loss_train: 1.6743 acc_train: 0.2000
Epoch: 4541 loss_train: 1.7419 acc_train: 0.2000
Epoch: 4561 loss_train: 1.6476 acc_train: 0.2000
Epoch: 4581 loss_train: 1.7262 acc_train: 0.2000
Final Test Acc: 0.2000  Best Acc: 0.6320 Pse_test acc: 0.4817 
Epoch: 4601 loss_train: 1.6614 acc_train: 0.2000
Epoch: 4621 loss_train: 1.6949 acc_train: 0.2000
Epoch: 4641 loss_train: 1.6772 acc_train: 0.2000
Epoch: 4661 loss_train: 1.6478 acc_train: 0.2000
Epoch: 4681 loss_train: 1.6744 acc_train: 0.2000
Final Test Acc: 0.2000  Best Acc: 0.6320 Pse_test acc: 0.5100 
Epoch: 4701 loss_train: 1.6476 acc_train: 0.2000
Epoch: 4721 loss_train: 1.6476 acc_train: 0.2000
Epoch: 4741 loss_train: 1.6687 acc_train: 0.2000
Epoch: 4761 loss_train: 1.6614 acc_train: 0.2000
Epoch: 4781 loss_train: 1.6635 acc_train: 0.2000
Final Test Acc: 0.2000  Best Acc: 0.6320 Pse_test acc: 0.4867 
Epoch: 4801 loss_train: 1.6342 acc_train: 0.2000
Epoch: 4821 loss_train: 1.6614 acc_train: 0.2000
Epoch: 4841 loss_train: 1.2431 acc_train: 0.4000
Epoch: 4861 loss_train: 1.6687 acc_train: 0.2000
Epoch: 4881 loss_train: 1.6688 acc_train: 0.2000
Final Test Acc: 0.2080  Best Acc: 0.6320 Pse_test acc: 0.4783 
Epoch: 4901 loss_train: 1.6355 acc_train: 0.2000
Epoch: 4921 loss_train: 1.6951 acc_train: 0.2000
Epoch: 4941 loss_train: 1.6943 acc_train: 0.2000
Epoch: 4961 loss_train: 1.6596 acc_train: 0.2000
Epoch: 4981 loss_train: 1.7262 acc_train: 0.2000
Final Test Acc: 0.2080  Best Acc: 0.6320 Pse_test acc: 0.4700 
Optimization Finished!
Total time elapsed: 30535.8223s
would you check these results, and give some advice?

Thank you.
